File: python/afordable_vacation.py
def find_min_cost(money, days, cost):
    total_days = len(cost)
    
    # Finding the minimum cost for exactly `days` consecutive days using sliding window
    current_sum = sum(cost[:days])
    min_cost = current_sum

    for i in range(days, total_days):
        current_sum += cost[i] - cost[i - days]
        min_cost = min(min_cost, current_sum)

    if min_cost <= money:
        return f"money: {min_cost}"

    # Finding the maximum duration within the given money using sliding window
    max_duration = 0
    current_sum = 0
    left = 0

    for right in range(total_days):
        current_sum += cost[right]

        while current_sum > money:
            current_sum -= cost[left]
            left += 1

        max_duration = max(max_duration, right - left + 1)

    return f"days: {max_duration}"

File: javascript/afordableVacation.js
function findMinCost(money, days, cost) {
    const totalDays = cost.length;
    
    // Finding the minimum cost for exactly `days` consecutive days using sliding window
    let currentSum = cost.slice(0, days).reduce((a, b) => a + b, 0);
    let minCost = currentSum;
  
    for (let i = days; i < totalDays; i++) {
      currentSum += cost[i] - cost[i - days];
      minCost = Math.min(minCost, currentSum);
    }
  
    if (minCost <= money) {
      return `money: ${minCost}`;
    }
  
    // Finding the maximum duration within the given money using sliding window
    let maxDuration = 0;
    currentSum = 0;
    let left = 0;
  
    for (let right = 0; right < totalDays; right++) {
      currentSum += cost[right];
  
      while (currentSum > money) {
        currentSum -= cost[left];
        left++;
      }
  
      maxDuration = Math.max(maxDuration, right - left + 1);
    }
  
    return `days: ${maxDuration}`;
  }
  